lib/Doctrine/DBAL/Cache/ResultCacheStatement.php

```diff
@@ -147,30 +146,38 @@ public function getIterator()
     /**
      * {@inheritdoc}
      */
-    public function fetch($fetchMode = null, $cursorOrientation = \PDO::FETCH_ORI_NEXT, $cursorOffset = 0)
+    public function fetch($fetchMode = null, $_ = null, $__ = null)
```

Comment:
We should. This particular commit should only go to `develop`. Initially, I wanted to get rid of them entirely but here's the trick:

1. `DBAL\PDOStatement` extends `PDOStatement` and implements `ResultStatement`, both of which declare `fetch()`.
2. It looks impossible to use a custom PDO statement class which doesn't extend `PDOStatement` since `PDO` instantiates it without calling constructor (this is my current understanding, I may be wrong).
3. Therefore, the declaration of `ResultStatement::fetch()` and all its implementors should be compatible with `PDOStatement::fetch()`.

I'd love to be able to rework `DBAL\PDOStatement extends PDOStatement` into a composition and get rid of those parameters entirely but it's technically challenging. Any ideas?